nuxeo-admin-console-web/angular-app/src/app/features/stream/components/consumer-thread-pool/consumer-thread-pool.component.ts

```diff
@@ -0,0 +1,347 @@
+        if (data?.length > 0) {
+          this.consumers = data;
+          this.selectedConsumer = this.consumers
+            ? this.consumers[0]?.consumer
+            : "";
+          this.streamForm.patchValue({
+            consumer: this.selectedConsumer,
+          });
+        }
+        this.isStartStopConsumerThreadBtnDisabled = !this.selectedConsumer
+          ? true
+          : false;
+      });
+
+    this.fetchConsumersError$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((error) => {
+        if (error) {
+          this.isStartStopConsumerThreadBtnDisabled = true;
+          this.showActionErrorModal({
+            type: ERROR_TYPES.SERVER_ERROR,
+            details: {
+              status: (error as HttpErrorResponse)?.status,
+              message: (error as HttpErrorResponse)?.message,
+            },
+          });
+        }
+      });
+
+    this.isStartConsumerThreadPoolSuccess$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((data: ConsumerThreadPoolState) => {
+        if (data) {
+          this.isStartOrStopConsumerThreadInProgress =
+            data?.isStartStopConsumerPoolProcessRunning;
+
+          if (data?.isStartProcessCompleted) {
+            this.streamService.showSuccessMessage(
+              CONSUMER_THREAD_POOL_LABELS.START_CONSUMER_SUCCESS_MSG
+            );
+          }
+        }
+      });
+
+    this.isStartConsumerThreadPoolFailure$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((error: unknown) => {
+        if (error) {
+          this.showActionErrorModal({
+            type: ERROR_TYPES.SERVER_ERROR,
+            details: {
+              status: (error as HttpErrorResponse)?.status,
+              message: (error as HttpErrorResponse)?.message,
+            },
+          });
+        }
+      });
+
+    this.isStopConsumerThreadPoolSuccess$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((data: ConsumerThreadPoolState) => {
+        if (data) {
+          this.isStartOrStopConsumerThreadInProgress =
+            data?.isStartStopConsumerPoolProcessRunning;
+        }
+
+        if (data?.isStopProcessCompleted) {
+          this.streamService.showSuccessMessage(
+            CONSUMER_THREAD_POOL_LABELS.STOP_CONSUMER_SUCCESS_MSG
+          );
+        }
+      });
+
+    this.isStopConsumerThreadPoolFailure$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((error: unknown) => {
+        if (error) {
+          this.showActionErrorModal({
+            type: ERROR_TYPES.SERVER_ERROR,
+            details: {
+              status: (error as HttpErrorResponse)?.status,
+              message: (error as HttpErrorResponse)?.message,
+            },
+          });
+        }
+      });
+  }
+
+  onStartStopConsumerThreadPool(operationType: string): void {
+    const params = {
+      consumer: this.streamForm?.get(CONSUMER_THREAD_POOL_LABELS.CONSUMER.LABEL)
+        ?.value,
+    };
+    if (
```

Comment:
instead of else-if, can we make this simply if and else ? there are only 2 possible values for operationType, right ?